src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -262,7 +264,10 @@ public CompiledScript compileInternal(String lang,  String script, ScriptType sc
             script = getScriptFromIndex(indexedScript.lang, indexedScript.id);
         }
 
-        CompiledScript compiled = cache.getIfPresent(cacheKey);
+        CompiledScript compiled = null;
+        if (! ignoreCache) {
+            cache.getIfPresent(cacheKey);
```

Comment:
That was supposed to read

compiled = cache.getIf...


Stupid typo on my side ...